skqulacs/qnn/regressor.py

```diff
@@ -186,12 +195,35 @@ def _cost_func_grad(
         for h in range(len(x_scaled)):
             backobs = Observable(self.n_qubit)
             if self.n_outputs >= 2:
-                for i in range(self.n_outputs):
+                for i in self.observables_str:
                     backobs.add_operator(
-                        (-y_scaled[h][i] + mto[h][i]) / self.n_outputs, f"Z {i}"
+                        2 * (-y_scaled[h][i] + mto[h][i]) / self.n_outputs,
+                        i,  # I add a 2* as a derivative of the RMSE error
```

Comment:
Why do you add a 2* ?
this sentence `a derivative of the RMSE error`, which i can't understand.
Please explanation detail.

Reply:
I think that since the RMSE is (-y_label+y_pred)^2, we need to account the derivative as 2*(-y_label+y_pred)dy_pred/dtheta. So that's why I multiplied for a 2. Is just a constant anyway.

Reply:
we times 2 at qualcs backprop. because you don't have to add *2..
https://github.com/qulacs/qulacs/blob/main/src/vqcsim/parametric_circuit.cpp#L274

Reply:
Ok, Thank you for the explanation. I will remove the 2*

Reply:
Upon verification, qualcs backprop was irrelevant. The original code was wrong, and *2 was correct.  Thank you.

Reply:
Ok I understand, I will correct then! Thank you for the feedback